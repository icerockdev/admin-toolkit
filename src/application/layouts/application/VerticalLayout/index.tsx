/* Copyright (c) 2020-2021 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license. */

import React, { FC } from 'react';
import { observer } from 'mobx-react';
import classNames from 'classnames';
import styles from './styles.module.scss';
import { useConfig } from '~/application/utils/hooks';
import { Link } from 'react-router-dom';
import { VerticalNavigation } from '~/containers/layout/VerticalNavigation';
import { Account } from '~/containers/auth/Account';

interface IProps {}

const VerticalLayout: FC<IProps> = observer(({ children }) => {
  const config = useConfig();

  return (
    <div className={classNames(styles.layout, 'vertical-layout')}>
      <div className={classNames(styles.menu, 'vertical-layout__menu')}>
        <div className={classNames(styles.logo, 'vertical-layout__logo')}>
          <Link to={config.fallbackUrl}>
            <img src={config.logo}  alt="logo"/>
          </Link>
        </div>

        <div
          className={classNames(
            styles.navigation,
            'vertical-layout__navigation'
          )}
        >
          <VerticalNavigation />
        </div>

        {!!config.auth && (
          <div
            className={classNames(styles.account, 'vertical-layout__account')}
          >
            {!!config.auth && <Account />}
          </div>
        )}
      </div>

      <div className={classNames(styles.content, 'vertical-layout__content')}>
        {children}
      </div>
    </div>
  );
});

export { VerticalLayout };
