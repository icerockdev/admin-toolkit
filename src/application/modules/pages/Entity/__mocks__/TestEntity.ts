import { Entity, IEntityProps } from "~/application";
import { TEST_FIELDS } from "~/application/modules/pages/Entity/__mocks__/testFields";
import { action, reaction } from "mobx";
import { getReferenceData } from "~/application/modules/pages/Entity/__mocks__/testApi";

class TestEntity extends Entity {
  constructor(fields?: Partial<IEntityProps>) {
    super(fields as any);
    reaction(
      () => this.editorData?.testReference,
      () => this.updateReferenceOptions()
    );
    this.references.testReference.getMany(this).then((result) =>
      Object.assign(this.referenceData, {testReference: result})
    )
  }

  @action
  updateReferenceOptions = async () => {
    if (!this.references.testReference.getMany) return;
    const results = await this.references.testReference.getMany(this);
    Object.assign(this.referenceData, {testReference: results});
  };
}

export default new TestEntity({
  title: 'TestEntity',
  editable: true,
  fields: TEST_FIELDS,
  references: {
    testReference: {
      getMany: (entity) => getReferenceData(entity),
    },
  },
});
