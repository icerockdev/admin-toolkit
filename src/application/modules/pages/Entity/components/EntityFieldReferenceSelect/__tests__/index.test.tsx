/* Copyright (c) 2021 IceRock MAG Inc. Use of this source code is governed by the Apache 2.0 license. */

import React from 'react';
import renderer from 'react-test-renderer';
import { EntityFieldReferenceSelect } from "~/application";
import { observable } from "mobx";
import TestEntity from "~/application/modules/pages/Entity/__mocks__/TestEntity";

test('EntityFieldReferenceSelect view test', () => {
  const element = <EntityFieldReferenceSelect
    name="testReference"
    value={1}
    fields={[]}
    entity={TestEntity}
  />

  const component = renderer.create(element);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('EntityFieldReferenceSelect edit test', () => {
  const store = new class {
    @observable value = 1;
  }();

  const handler = (value) => store.value = value

  const element = <EntityFieldReferenceSelect
    name="testReference"
    value={store.value}
    fields={[]}
    handler={handler}
    isEditing={true}
    entity={TestEntity}
  />

  const component = renderer.create(element);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
