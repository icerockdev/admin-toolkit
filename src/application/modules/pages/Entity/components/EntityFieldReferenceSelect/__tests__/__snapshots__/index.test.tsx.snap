// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntityFieldReferenceSelect edit test 1`] = `
<div
  className="MuiFormControl-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
    data-shrink={true}
    style={
      Object {
        "whiteSpace": "nowrap",
      }
    }
  />
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
    style={
      Object {
        "minWidth": 40,
      }
    }
  >
    <div
      aria-haspopup="listbox"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input select"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      variant 1
    </div>
    <input
      aria-hidden={true}
      className="MuiSelect-nativeInput"
      onAnimationStart={[Function]}
      onChange={[Function]}
      required={false}
      tabIndex={-1}
      value={1}
    />
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
      style={
        Object {
          "paddingLeft": 8,
        }
      }
    >
      <legend
        className="PrivateNotchedOutline-legend-2"
        style={
          Object {
            "width": 0.01,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`EntityFieldReferenceSelect view test 1`] = `
<div>
  variant 1
</div>
`;
